### ISSUE TITLE
tokio-runtime-worker panicked

### ISSUE THREAD

**Issue by deni64k** [sigp/lighthouse#2099]
## Description

Hello,

We see this error after which Lighthouse paralyzes. No backtrace, unfortunately.

```
lighthouse[224353]: Dec 16 23:22:43.234 INFO New block received                      hash: 0xd161…16bc, slot: 205013
lighthouse[224353]: Dec 16 23:22:49.000 INFO Synced                                  slot: 205013, block: 0xd161…16bc, epoch: 6406, finalized_epoch: 6404, finalized_root: 0x6fa5…4666, peers: 513, service: slot_notifier
lighthouse[224353]: Dec 16 23:22:56.086 INFO New block received                      hash: 0x7d77…413c, slot: 205014
lighthouse[224353]: Dec 16 23:23:01.005 INFO Synced                                  slot: 205014, block: 0x7d77…413c, epoch: 6406, finalized_epoch: 6404, finalized_root: 0x6fa5…4666, peers: 515, service: slot_notifier
lighthouse[224353]: Dec 16 23:23:07.359 INFO New block received                      hash: 0x7cde…2c10, slot: 205015
lighthouse[224353]: thread 'tokio-runtime-worker' panicked at 'elapsed=37085312; when=37085312', /opt/rust/registry/src/github.com-1ecc6299db9ec823/tokio-util-0.5.1/src/time/wheel/mod.rs:245:5
lighthouse[224353]: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
lighthouse[224353]: Dec 16 23:23:13.000 INFO Synced                                  slot: 205015, block: 0x7cde…2c10, epoch: 6406, finalized_epoch: 6404, finalized_root: 0x6fa5…4666, peers: 514, service: slot_notifier
lighthouse[224353]: Dec 16 23:23:15.009 WARN Error processing HTTP API request       method: POST, path: /eth/v1/validator/aggregate_and_proofs, status: 500 Internal Server Error, elapsed: 6.123981ms
lighthouse[224353]: Dec 16 23:23:15.010 WARN Error processing HTTP API request       method: POST, path: /eth/v1/validator/aggregate_and_proofs, status: 500 Internal Server Error, elapsed: 7.22892ms
lighthouse[224353]: Dec 16 23:23:23.036 WARN Error processing HTTP API request       method: POST, path: /eth/v1/beacon/pool/attestations, status: 500 Internal Server Error, elapsed: 1.734088ms
lighthouse[224353]: Dec 16 23:23:23.047 WARN Error processing HTTP API request       method: POST, path: /eth/v1/beacon/pool/attestations, status: 500 Internal Server Error, elapsed: 1.697393ms
lighthouse[224353]: Dec 16 23:23:23.056 WARN Error processing HTTP API request       method: POST, path: /eth/v1/beacon/pool/attestations, status: 500 Internal Server Error, elapsed: 1.653093ms
lighthouse[224353]: Dec 16 23:23:23.080 WARN Error processing HTTP API request       method: POST, path: /eth/v1/beacon/pool/attestations, status: 500 Internal Server Error, elapsed: 2.599251ms
lighthouse[224353]: Dec 16 23:23:23.100 WARN Error processing HTTP API request       method: POST, path: /eth/v1/beacon/pool/attestations, status: 500 Internal Server Error, elapsed: 1.501371ms
lighthouse[224353]: Dec 16 23:23:25.000 INFO Synced                                  slot: 205016, block:    …  empty, epoch: 6406, finalized_epoch: 6404, finalized_root: 0x6fa5…4666, peers: 514, service: slot_notifier
```

## Version

```
$ lighthouse --version
Lighthouse v1.0.4-1abc70e
BLS Library: blst
Specs: mainnet (true), minimal (false), v0.12.3 (false)
```

```
$ CARGO_HOME=/opt/rust RUSTUP_HOME=/opt/rust RUSTUP_TOOLCHAIN=stable /opt/rust/bin/rustc --version
rustc 1.48.0 (7eac88abb 2020-11-16)
```

## Present Behaviour

Lighthouse Beacon Node stops any network activity and doesn't submit attestations.

## Expected Behaviour

Lighthouse Beacon Node performs network activity and submits attestations.

## Steps to resolve

We simply restarted the node.

**Comment by realbigsean** [comment]
Seems related to this issue: #1067

> peers: 515
I'd guess it's got something to do with the very high peer count

**Comment by AgeManning** [comment]
Yes. The previous issue was in the delay queues managing the ping and meta-data intervals. There must still be an issue in tokio when these are pushed with very high peer counts. 

It looks to me like it is currently being addressed in tokio by this PR: https://github.com/tokio-rs/tokio/pull/3270

Hopefully we can update tokio once that gets merged to resolve this issue. It seems to be me its currently probabilistic in hitting specific delay timeouts and that probability is exasperated by large peer counts. 

An interim solution would be to lower the peer count to reduce the chances of hitting this. In the meantime, I think we need to wait for the upstream fix.

**Comment by paulhauner** [comment]
Tokio https://github.com/tokio-rs/tokio/pull/3270 has been merged into their `master` so I assume we'll see a release soon.

**Comment by paulhauner** [comment]
It seems [`tokio_util`](https://crates.io/crates/tokio-util) was updated to `0.6.2` and that contains https://github.com/tokio-rs/tokio/pull/3270.

**Comment by paulhauner** [comment]
Hopefully resolved in #2172, which includes `tokio-util 0.6.3`.
